ambry-store/src/test/java/com.github.ambry.store/StoreFindTokenTest.java

```diff
@@ -168,35 +165,54 @@ public void constructionErrorCasesTest() {
     } catch (IllegalArgumentException e) {
       // expected. Nothing to do.
     }
-
-    // no offset in IndexBased
-    try {
-      new StoreFindToken(key, null, sessionId, null);
-      fail("Construction of StoreFindToken should have failed");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // no sessionId in IndexBased
-    try {
-      new StoreFindToken(key, offset, null, null);
-      fail("Construction of StoreFindToken should have failed");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
   }
 
   // helpers
   // general
 
   /**
-   * Compares two tokens to ensure their equality test passes and that their hash codes match.
+   * Compares two tokens to ensure their equality test passes and that their hash codes matches for recent
+   * versions
+   * @param reference the reference {@link StoreFindToken}
+   * @param toCheck the {@link StoreFindToken} to check
+   * @param version the StoreFindToken version to be tested
+   */
+  private void compareTokens(StoreFindToken reference, StoreFindToken toCheck, short version) {
+    switch (version) {
```

Comment:
lets say we create a Journal Based StoreFindToken with inclusive set to true. During serialization in V_1 (check out getSerializedStream() at line 239 in StoreFindTokenTest), we don't serialize the inclusiveness property. So, the deserailized token too doesn't have the inclusiveness property set.